public/pages/ConfigureModel/components/CategoryField/__tests__/CategoryField.test.tsx

```diff
@@ -227,4 +225,137 @@ describe('<CategoryField /> spec', () => {
+      <Fragment>
+        <Formik
+          initialValues={{
+            categoryField: [],
+          }}
+          onSubmit={() => {}}
+        >
+          <Fragment>
+            <Form>
+              <CategoryField
+                isEdit={true}
+                isHCDetector={true}
+                categoryFieldOptions={['a', 'b', 'c']}
+                setIsHCDetector={(isHCDetector: boolean) => {
+                  return;
+                }}
+                isLoading={false}
+                formikProps={{
+                  values: {
+                    categoryFieldEnabled: true,
+                  },
+                }}
+              />
+            </Form>
+          </Fragment>
+        </Formik>
+      </Fragment>
+    );
+    // try to open combo box. options should not show since the box is readonly
+    fireEvent.click(getByTestId('comboBoxToggleListButton'));
+    expect(queryByText('a')).toBeNull();
+    expect(queryByText('b')).toBeNull();
+    expect(queryByText('c')).toBeNull();
+  });
+  test('shows warning callout if creating & category field enabled', () => {
+    const { queryByTestId } = render(
+      <Fragment>
+        <Formik
+          initialValues={{
+            categoryField: [],
+          }}
+          onSubmit={() => {}}
+        >
+          <Fragment>
+            <Form>
+              <CategoryField
+                isEdit={false}
+                isHCDetector={true}
+                categoryFieldOptions={[]}
+                setIsHCDetector={(isHCDetector: boolean) => {
+                  return;
+                }}
+                isLoading={false}
+                formikProps={{
+                  values: {
+                    categoryFieldEnabled: true,
+                  },
+                }}
+              />
+            </Form>
+          </Fragment>
+        </Formik>
+      </Fragment>
+    );
+    expect(queryByTestId('cannotEditCategoryFieldCallout')).not.toBeNull();
+  });
+  test('hides warning callout if creating & category field disabled', () => {
+    const { queryByTestId } = render(
+      <Fragment>
+        <Formik
+          initialValues={{
+            categoryField: [],
+          }}
+          onSubmit={() => {}}
+        >
+          <Fragment>
+            <Form>
+              <CategoryField
+                isEdit={false}
+                isHCDetector={true}
+                categoryFieldOptions={[]}
+                setIsHCDetector={(isHCDetector: boolean) => {
+                  return;
+                }}
+                isLoading={false}
+                formikProps={{
+                  values: {
+                    categoryFieldEnabled: true,
+                  },
+                }}
+              />
+            </Form>
+          </Fragment>
+        </Formik>
+      </Fragment>
+    );
+    expect(queryByTestId('cannotEditCategoryFieldCallout')).not.toBeNull();
+  });
+  test('shows info callout and hides warning callout if editing', () => {
```

Comment:
The test looks exactly the same with the last one.  Anything I am missing?

Reply:
The one on lines 296-327 are for testing that the warning callout doesn't show unless the user enables category fields (the first screenshot in description)

The one on lines 328-360 is for testing that the info callout shows (the second screenshot in description), and that the warning callout is hidden

Reply:
so the only difference between lines 296-327 and lines 328-360 is that the 2nd test case checked one more condition "expect(queryByTestId('cannotEditCategoryFieldCallout')).toBeNull();"? If not, what are the differences in code otherwise?

Reply:
Oh, I think I see your original comment was probably about the test on lines 264-295, vs, the test on lines 296-327. That's a good catch - need to change the test on lines 296-327. Updating now.

Reply:
Fixed here: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/79/commits/d27a7dd10133f8272f05e4044d0402a062ea1ac5